src/Plugins/RestServer/Controllers/v1/ContractsController.cs

```diff
@@ -0,0 +1,220 @@
+using Neo.Plugins.RestServer.Models.Error;
+using Neo.SmartContract;
+using Neo.SmartContract.Manifest;
+using Neo.SmartContract.Native;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+
+namespace Neo.Plugins.RestServer.Controllers.v1
+{
+    [Route("/api/v{version:apiVersion}/contracts")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ErrorModel))]
+    [ApiVersion("1.0")]
+    [ApiController]
+    public class ContractsController : ControllerBase
+    {
+        private readonly NeoSystem _neosystem;
+
+        public ContractsController()
+        {
+            _neosystem = RestServerPlugin.NeoSystem ?? throw new NodeNetworkException();
+        }
+
+        /// <summary>
+        /// Get all the smart contracts from the blockchain.
+        /// </summary>
+        /// <param name="skip" example="1">Page</param>
+        /// <param name="take" example="50">Page Size</param>
+        /// <returns>An array of Contract object.</returns>
+        /// <response code="204">No more pages.</response>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet(Name = "GetContracts")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ContractState[]))]
+        public IActionResult Get(
+            [FromQuery(Name = "page")]
+            int skip = 1,
+            [FromQuery(Name = "size")]
+            int take = 50)
+        {
+            if (skip < 1 || take < 1 || take > RestServerSettings.Current.MaxPageSize)
+                throw new InvalidParameterRangeException();
+            var contracts = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView);
+            if (contracts.Any() == false)
+                return NoContent();
+            var contractRequestList = contracts.OrderBy(o => o.Manifest.Name).Skip((skip - 1) * take).Take(take);
+            if (contractRequestList.Any() == false)
+                return NoContent();
+            return Ok(contractRequestList);
+        }
+
+        /// <summary>
+        /// Gets count of total smart contracts on blockchain.
+        /// </summary>
+        /// <returns>Count Object</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("count", Name = "GetContractCount")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(CountModel))]
+        public IActionResult GetCount()
+        {
+            var contracts = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView);
+            return Ok(new CountModel() { Count = contracts.Count() });
+        }
```

Comment:
```suggestion
        /// <summary>
        /// Get all the smart contracts in range.
        /// </summary>
        /// <param name="start" example="1">start index</param>
        /// <param name="end" example="50">end index</param>
        /// <returns>An array of Contract object.</returns>
        /// <response code="204">No more pages.</response>
        /// <response code="200">Successful</response>
        /// <response code="400">An error occurred. See Response for details.</response>
        [HttpGet("range", Name = "GetContractRange")]
        [ProducesResponseType(StatusCodes.Status204NoContent)]
        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ContractState[]))]
        public IActionResult GetContractRange(
            [FromQuery(Name = "start")]
            int start = 1,
            [FromQuery(Name = "end")]
            int end = 50)
        {
            if (start < 1 || end < 1 || end > RestServerSettings.Current.MaxPageSize)
                throw new InvalidParameterRangeException();
            var contracts = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView);
            if (contracts.Any() == false)
                return NoContent();
            var contractRequestList = contracts.OrderBy(o => o.Id).Take(new Range(start - 1, end));
            if (contractRequestList.Any() == false)
                return NoContent();
            return Ok(contractRequestList);
        }

        /// <summary>
        /// Gets count of total smart contracts on blockchain.
        /// </summary>
        /// <returns>Count Object</returns>
        /// <response code="200">Successful</response>
        /// <response code="400">An error occurred. See Response for details.</response>
        [HttpGet("count", Name = "GetContractCount")]
        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(CountModel))]
        public IActionResult GetCount()
        {
            var contracts = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView);
            return Ok(new CountModel() { Count = contracts.Count() });
        }
```
Can we add a new `contracts/range` API? It should be useful.

Reply:
Just use `/api/v1/contracts?page={number}&size={number}`